app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -1,33 +1,124 @@
 package com.example.taskyy.ui.viewmodels
 
 import android.content.SharedPreferences
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import com.example.taskyy.ui.events.ReminderEvent
+import com.example.taskyy.ui.objects.Day
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Locale
 import javax.inject.Inject
 
 @HiltViewModel
 class ReminderViewModel @Inject constructor(
-    private val sharedPreferences: SharedPreferences
+    private val sharedPreferences: SharedPreferences,
```

Comment:
Here, you're coupling a data related dependency to the presentation layer.

The following allows you to easily access user data from any layer in your app while being able to also easily swap out preferences implementations. The only thing needed for that would be to swap out the shared preferences object in the `UserPreferencesImpl` constructor with let's say a DataStore instance.

*Domain model*
```
data class AuthenticatedUser(
    val fullName: String,
    val email: String,
    val token: String
)
```

*Domain abstraction*
```
interface UserPreferences {
    fun saveAuthenticatedUser(user: AuthenticatedUser)
    fun getAuthenticatedUser(): AuthenticatedUser?
    fun clearPreferences()
}
```

*Data implementation*
```
class UserPreferencesImpl(
    private val sharedPref: SharedPreferences
): UserPreferences {
    override fun saveAuthenticatedUser(user: AuthenticatedUser) {
        // Save JSON of user in shared pref
    }

    override fun getAuthenticatedUser(): AuthenticatedUser? { ... }

    override fun clearPreferences() { ... }
}
```

In your Dagger module:
```
@Provides
@Singleton
fun provideUserPreferences(sharedPreferences: SharedPreferences): UserPreferences {
    return UserPreferencesImpl(sharedPreferences)
}
```

Then, for example in any ViewModel:
```
@HiltViewModel
class MyViewModel @Inject constructor(
    private val userPreferences: UserPreferences
): ViewModel() {
    // Use userPreferences
}
```